server/webSocketServer/index.js

```diff
@@ -0,0 +1,33 @@
+const WebSocket = require('ws');
+const userHolder = require('./userHolder');
+const messageCreator = require('./messageCreator');
+
+const startWebSocketServer = (httpServer) => {
+
+    const webSocketServer = new WebSocket.Server({ server: httpServer });
+    const users = userHolder();
+
+    webSocketServer.on('connection', webSocketConnection => {
+        const userId = users.addUser(webSocketConnection);
+        const user = users.getUser(userId);
+
+        webSocketConnection.on('message', rawMessage => {
+            const parsedMessage = JSON.parse(rawMessage);
+            const messageToSend = messageCreator.createMessage(parsedMessage.data, user);
```

Comment:
what if parsedMessage will be undefined ??

Reply:
No good

Reply:
fixed

Reply:
nice